app/src/main/kotlin/com/mapzen/erasermap/presenter/MainPresenterImpl.kt

```diff
@@ -222,6 +229,25 @@ open class MainPresenterImpl(val mapzenLocation: MapzenLocation, val bus: Bus,
   }
 
   @Subscribe fun onRoutePreviewEvent(event: RoutePreviewEvent) {
+    if (locationClientManager.getClient() == null) {
+      locationClientManager.connect()
+      locationClientManager.addRunnableToRunOnConnect(
+          Runnable { onRoutePreviewEvent(event) }
+      )
+      return
+    }
+
+    val locationStatusCode = locationSettingsChecker.getLocationStatusCode(mapzenLocation,
+        locationClientManager)
+    when (locationStatusCode) {
```

Comment:
Maybe better to use an `if` statement here? `when` in Kotlin is kinda like `switch` in Java. This works but feels a little weird to only have one case.